isyfact-standards-doc/src/docs/antora/modules/isy-security/pages/konzept/inhalt.adoc

```diff
@@ -226,6 +226,75 @@ OpenID Connect Abfragen zur Identität eines Benutzers, OpenID-Token oder Custom
+Die bisher aufgezeigten Authentifizierungswege werden in Bezug auf `isy-security` als jeweilig unterschiedliche Mandanten behandelt bzw. konfiguriert.
+Aber auch pro Authentifizierungsweg ist es möglich unterschiedliche Mandanten sprich mehrere Instanzen des gleichen Typs einzurichten.
+
+Um nun für jeden Mandanten eine eindeutige Zuordnung bzw. eine Mandanten spezifische Gruppierung der Properties herzustellen, muss der Property-Key im Pfad den Namen des Mandanten beinhalten:
+
+`spring.security.oauth2.resourceserver.jwt.tenants.<mandant-id>`
+
+Der Platzhalter `<mandant-id>` wird in der konkreten Konfigurationsausprägung auf den Namen des Mandanten gesetzt.
+So wäre beispielsweise für die drei Authentifizierungswege aus der Nutzungsschicht folgende Property-Pfade vorstellbar:
+
+- `spring.security.oauth2.resourceserver.jwt.tenants.*portal*`
+- `spring.security.oauth2.resourceserver.jwt.tenants.*sgw*`
+- `spring.security.oauth2.resourceserver.jwt.tenants.*agw*`
+
+Sollten für einen Authentifizierungskanal mehrere Instanzen notwendig werden, dann müssen sich deren Namen natürlich auch unterscheiden, wie beispielsweise:
+
+- `spring.security.oauth2.resourceserver.jwt.tenants.*sgw-1*`
+- `spring.security.oauth2.resourceserver.jwt.tenants.*sgw-2*`
+
+
+===== Mandanten spezifische Konfiguration am Beispiel Portal
+So bedarf beispielsweise die Authentifizierung einer externen WebClient-Anwendung über das _Portal_-Gateway eine Konfiguration des IAM-Service, die berücksichtigt, dass der Aufruf des IAM-Service hier eine Unterscheidung zwischen _internem_ und _externem_ Aufruf benötigt. Wie und wo diese Aufruf-Unterscheidung zum Tragen kommt, zeigt nun das nachfolgende Bild die Kommunikationswege auf, die ein WebClient über ein Portal zur Authentifizierung und Autorisierung durchläuft.
+
+.Kommunikationswege einer WebClient-Authentifizierung über Portal
+[id="image-webclient-portal_authentication",reftext="{figure-caption} {counter:figures}"]
+image::isy-security:konzept/WebClient_Portal_Authentication.dn.svg[align="center"]
+
+Schrittweise Erklärung des Kommunikationsablaufs:
+
+- Eine Anwendung in Anwendungslandschaft AWL-x startet einen WebClient
+- (1) Dieser für AWL-1 externe WebClient sendet seine Requests an das _Portal_ in der AWL-1
+- (2) Das _Portal_ (Apache tomcat mit _openid-connect_ Modul) erkennt die ausstehende Authentifizierung und
+- (3) liefert an den externen WebClient einen Redirect „Login mit der *externen* Issuer-URI“, um (4) direkt im IAM/KeyCloak der AWL-1 den (5) LoginDialog zu öffnen.
+- (6) In diesem Dialog gibt der User des WebClients sein _Username_ und _Passwort_ ein, aus dem im _IAM/KeyCloak_ bei valider Authentifizierung ein _OneTime AutorisationCode_ erzeugt und (7) als Redirect an den _WebClient_ zurückgegeben wird.
+- (8) Den Redirect mit dem _AutorisationCode_ sendet der _WebClient_ wieder an das _Portal_, das (9) diesen über die *interne* Issuer-URI an den _IAM/KeyCloak_ in der AWL-1 weiterleitet.
```

Comment:
OK, das bedeutet, dass der IAM/KeyCloak die Callback-Adresse des Portals (eigentlich: mod_auth_openidc-Modul) kennen muss. KeyCloak: Client- Client details (des confidential client 'portal'): Access settings - Home URL (Default URL to use when the auth server needs to redirect or link back to the client.)